lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -567,6 +569,25 @@ def get(self, request, course_key):
             serializer = StudentGradebookEntrySerializer(entries, many=True)
             return self.get_paginated_response(serializer.data, **users_counts)
 
+    def _get_user_count(self, query_args, cache_time=3600):
+        """
+        Return the user count for the given query arguments to CourseEnrollment.
+
+        caches the count for cache_time seconds.
+        """
+        hasher = hashlib.md5()
+        for arg in query_args:
+            hasher.update(bytes(arg))
```

Comment:
I am curious why is it better to use hasher vs just join all the query args to a string and use that as a key?

Reply:
Because the string version of the combined arguments could be very long. Max key size for memcached is 250 bytes.

Reply:
`edx-platform` already does this automatically: https://github.com/edx/edx-platform/blob/master/common/djangoapps/util/memcache.py#L32